src/state.rs

```diff
@@ -84,16 +84,23 @@ impl ProgressState {
     }
 
     /// Return the current average time per step
+    #[allow(dead_code)] // Keep for API compatibility. Should deprecate.
```

Comment:
I think I'd like to just keep this but also add some explanation to the docs here about the limitations of `Duration` resolution.

Reply:
What might be a reason to keep Duration around? It might be a source of bugs for users of the crate, similar to the bugs this PR fixes.

Reply:
I think that `Duration` as a type is easier to work with than a floating point type: it has a bunch of facilities that cover common use cases for dealing with durations, and it's more obvious to understand when you have it in your code. There are definitely use cases where the 1ns resolution is not an issue, so those users might like to get the more obvious type.

Reply:
I'm definitely not going to argue about this much further and it's not a hill I'm willing to die on, but it _seems_ that average time per step can very easily get <1ns when dealing with bytes. Surfacing this to the user of the library allows them to introduce a bug that won't be caught at compile time and may not even happen until their application runs on a sufficiently fast machine (i.e. reading files from SATA vs NVMe SSD or using 1 vs 10 GbE). I'll be happy with just the minimal fix that allows built-in estimates and draw rate throttling functionality to work in my application :)